.github/workflows/test.yml

```diff
@@ -28,46 +28,62 @@ jobs:
     - name: install additional required packages
       run: |
         sudo apt-get install -y --no-install-recommends \
-        libudunits2-dev \
-        libproj-dev \
-        libgeos-dev \
-        libffi-dev \
-        cmake \
-        libgtest-dev \
-        build-essential \
-        curl \
-        libcurl4-openssl-dev
+          libudunits2-dev \
+          gdal-bin libgdal-dev \
+          libproj-dev \
+          libgeos-dev \
+          libffi-dev \
+          cmake \
+          libgtest-dev \
+          build-essential \
+          curl \
+          libcurl4-openssl-dev \
+          python3-numpy \
+          python3-shapely \
+          python3-cartopy
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
+    - name: Install dependencies (inline, no external reqs)
       run: |
-        python -m pip install --upgrade pip==24.0
+        # 1. Pin pip & setuptools so distutils & flags work
+        python -m pip install --upgrade pip==24.0 setuptools==65.5.1 wheel setuptools_scm
 
-        # patch to solve dependency issues with celery / fiona / click / pytz
-        # download celery package 
+        # 2. Point pip at your IMOS PyPI repo (so aodntools>=1.3.1 is found)
+        export PIP_INDEX_URL="http://imos-artifacts.s3-website-ap-southeast-2.amazonaws.com/repo/pypi/${STAGE}/"
```

Comment:
is this wookring? STAGE is an VENV, but then "stage" is defined above. not the same case?